main.nf

```diff
@@ -1,114 +1,26 @@
 #!/usr/bin/env nextflow
-
-/// To use DSL-2 will need to include this
 nextflow.enable.dsl=2
+params.fq = "*.fq.gz"
 
-// =================================================================
-// main.nf is the pipeline script for a nextflow pipeline
-// Should contain the following sections:
-	// Process definitions
-    // Channel definitions
-    // Workflow structure
-	// Workflow summary logs 
-
-// Examples are included for each section. Remove them and replace
-// with project-specific code. For more information see:
-// https://www.nextflow.io/docs/latest/index.html.
-//
-// ===================================================================
-
-// Import processes or subworkflows to be run in the workflow
-// Each of these is a separate .nf script saved in modules/ directory
-// See https://training.nextflow.io/basic_training/modules/#importing-modules 
-include { processOne } from './modules/process1'
-include { processTwo } from './modules/process2' 
-
-// Print a header for your pipeline 
-log.info """\
-
-=======================================================================================
-Name of the pipeline - nf 
-=======================================================================================
-
-Created by <YOUR NAME> 
-Find documentation @ https://sydney-informatics-hub.github.io/Nextflow_DSL2_template_guide/
-Cite this pipeline @ INSERT DOI
-
-=======================================================================================
-Workflow run parameters 
-=======================================================================================
-input       : ${params.input}
-outDir      : ${params.outDir}
-workDir     : ${workflow.workDir}
-=======================================================================================
-
-"""
-
-/// Help function 
-// This is an example of how to set out the help function that 
-// will be run if run command is incorrect or missing. 
-
-def helpMessage() {
-    log.info"""
-  Usage:  nextflow run main.nf --input <samples.tsv> 
+process fastqc {
 
-  Required Arguments:
+    cpus 2
+    memory '12 GB'
+    container "/home/ubuntu/singularity-hpc"
```

Comment:
how does this link to the fastqc container? Could you instead specify more like [this](https://github.com/Sydney-Informatics-Hub/IndexReferenceFasta-nf/blob/b1de54a594e2f11295c9cd41be8b03d991bd4883/modules/gatk.nf#L4)

Reply:
search for the fastqc biocontainer in https://quay.io/ and provide to container scope i.e:
```
container "quay.io/biocontainers/fastqc:0.12.1--hdfd78af_0" 
```

* Nextflow will pull a container for you automatically when you run a workflow
* TODO later look at what to do about pulling containers when your job has no external network access 
* When using Singularity to manage containers with nextflow, dont need to clarify it is a docker container. For example when pulling a Docker container with Singularity normally you would use: 
```
singularity pull docker://quay.io/biocontainers/fastqc:0.12.1--hdfd78af_0
```